core/src/main/java/bio/terra/pearl/core/service/dataimport/ImportItemService.java

```diff
@@ -3,20 +3,57 @@
 import bio.terra.pearl.core.dao.dataimport.ImportItemDao;
 import bio.terra.pearl.core.model.dataimport.Import;
 import bio.terra.pearl.core.model.dataimport.ImportItem;
+import bio.terra.pearl.core.model.dataimport.ImportItemStatus;
 import bio.terra.pearl.core.service.CrudService;
+import bio.terra.pearl.core.service.exception.NotFoundException;
+import bio.terra.pearl.core.service.participant.EnrolleeService;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
 
 @Service
 @Slf4j
 public class ImportItemService extends CrudService<ImportItem, ImportItemDao> {
 
+    @Autowired
+    EnrolleeService enrolleeService;
+
     public ImportItemService(ImportItemDao dao) {
         super(dao);
     }
 
     public void attachImportItems(Import dataImport) {
-        dataImport.setImportItems(dao.findByItem(dataImport.getId()));
+        dataImport.setImportItems(dao.findAllByImport(dataImport.getId()));
     }
 
+    @Transactional
+    public ImportItem updateStatus(UUID id, ImportItemStatus status) {
+        ImportItem importItem = dao.find(id).orElseThrow(() -> new NotFoundException("Import Item not found for id: " + id));
+        importItem.setStatus(status);
+        return dao.update(importItem);
+    }
+
+    @Transactional
+    public void updateStatusByImportId(UUID importId, ImportItemStatus status) {
+        List<ImportItem> importItems = dao.findAllByImport(importId);
+        importItems.forEach(importItem -> {
+            importItem.setStatus(status);
+            dao.update(importItem);
+        });
```

Comment:
since this is applying the same status change to all import items of the same importId, this could be done in one query. but im not sure that performance here is critical, so I'll leave it up to you

Reply:
I did consider it and ended up doing this way because I can use available code and performance critical